docs/schema-definition.md

```diff
@@ -105,63 +105,80 @@ $schema = new Schema($config);
+
+It is recommended to centralize this kind of functionality in a type loader. A typical example might look like the following:
 
 ```php
-use GraphQL\Type\Definition\Type;
-use GraphQL\Type\Definition\ObjectType;
-use GraphQL\Type\Schema;
 
+// MyAType.php
+class MyAType extends \GraphQL\Type\Definition\ObjectType {
+    public __construct() {
+        parent::__construct([
+            "name"=>"MyA",
+            "fields" => fn() => [
+                'b' => Types::MyB()
+        ]);
+    }
+}
+
+// MyBType.php
+class MyBType extends \GraphQL\Type\Definition\ObjectType {
+    public __construct() {
+        parent::__construct([
+            "name"=>"MyB",
+            "fields" => fn() => [
+                'c' => Type::string()
+        ]);
+    }
+}
+
+// TypeRegistry.php
 class TypeRegistry
 {
     /**
-     * @var array<string, Type>
+     * @var array<string, Type&NamedType>
      */
-    private array $types = [];
+    private static array $types = [];
 
-    public function get(string $name): Type
+    private static function get(string $classname): \Closure
     {
-        return $this->types[$name] ??= $this->{$name}();
+        return static fn () => self::byClassName($classname);
     }
 
-    private function MyTypeA(): ObjectType
+    private static function byClassName(string $classname): Type
     {
-        return new ObjectType([
-            'name' => 'MyTypeA',
-            'fields' => fn() => [
-                'b' => [
-                    'type' => $this->get('MyTypeB')
-                ],
-            ]
-        ]);
-    }
+        $parts = \explode('\\', $classname);
+        $cacheName = \strtolower(\preg_replace('~Type$~', '', $parts[\count($parts) - 1]));
 
-    private function MyTypeB(): ObjectType
-    {
-        // ...
+        return self::$types[$cacheName] ??= new $classname;
     }
+
+    public static function MyA(): callable { return self::get(MyAType::class); }
+    public static function MyB(): callable { return self::get(MyBType::class); }
+
+    ...
 }
 
+// ...
+
 $typeRegistry = new TypeRegistry();
 
 $schema = new Schema([
-    'query' => $typeRegistry->get('Query'),
-    'typeLoader' => static fn (string $name): Type => $typeRegistry->get($name),
+    'query' => new Query(),
+    'mutation' => new Mutation(),
+    'typeLoader' => static fn (string $name): Type =>
+        $typeRegistry->get($name),
 ]);
 ```
 
-You can automate this registry if you wish to reduce boilerplate or even
-introduce a Dependency Injection Container if your types have other dependencies.
-
-Alternatively, all methods of the registry could be static - then there is no need
-to pass it in the constructor - instead use **TypeRegistry::myAType()** in your type definitions.
+A working demonstration of this kind of architecture can be found in the [01-blog](../examples/01-blog) sample.
```

Comment:
Good, because this example is not working at all 😜

Reply:
Well, to be fair I'm not sure the old one worked, either (it references "Query" on  `$typeRegistry`).

Reply:
By the way, will this relative link work once the documentation site gets built? Should I use an absolute link?